services/wallet/collectibles/manager.go

```diff
@@ -1027,3 +1028,78 @@ func (o *Manager) SearchCollections(ctx context.Context, chainID walletCommon.Ch
 	}
 	return nil, ErrNoProvidersAvailableForChainID
 }
+
+func (o *Manager) FetchCollectionSocialsAsync(uniqueID thirdparty.CollectibleUniqueID) error {
+	hasValidSocials, err := o.collectionsDataDB.HasValidSocialsForID(uniqueID.ContractID)
+	if err != nil {
+		log.Debug("FetchCollectionSocialsAsync failed for", "chainID", uniqueID.ContractID.ChainID, "uniqueID", uniqueID, "err", err)
+		return err
+	}
+
+	if !hasValidSocials {
+		go func() {
+			defer o.checkConnectionStatus(uniqueID.ContractID.ChainID)
+
+			provider, socials, err := o.fetchSocialsForCollectibleUniqueID(context.Background(), uniqueID.ContractID.ChainID, uniqueID)
+			if err != nil || socials == nil {
+				log.Debug("FetchCollectionSocialsAsync failed for", "chainID", uniqueID.ContractID.ChainID, "uniqueID", uniqueID, "err", err)
+				return
+			}
+
+			socialsMessage := CollectibleSocialsMessage{
+				ID:      uniqueID,
+				Socials: socials,
+			}
+
+			err = o.collectionsDataDB.SetCollectionSocialsData(provider, uniqueID.ContractID, socials)
+			if err != nil {
+				log.Error("Error saving socials to DB: %v", err)
+				return
+			}
+
+			payload, err := json.Marshal(socialsMessage)
+			if err != nil {
+				log.Error("Error marshaling response: %v", err)
+				return
+			}
+
+			event := walletevent.Event{
+				Type:    EventGetCollectibleSocialsDone,
+				Message: string(payload),
+			}
+
+			o.feed.Send(event)
+		}()
+
+	}
+	return nil
+}
+
+func (o *Manager) fetchSocialsForCollectibleUniqueID(ctx context.Context, chainID walletCommon.ChainID, idToFetch thirdparty.CollectibleUniqueID) (string, *thirdparty.CollectionSocials, error) {
```

Comment:
Same, should be called "fetchSocialsForCollection" and take a ContractID. Also chainID is part of ContractID, no need to pass it separately